pyexchange/binance_us.py

```diff
@@ -0,0 +1,267 @@
+# This file is part of Maker Keeper Framework.
+#
+# Copyright (C) 2020 Exef
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from pprint import pformat, pprint
+from pyexchange.api import PyexAPI
+import hmac
+import hashlib
+import time
+import base64
+import requests
+import json
+
+import dateutil.parser
+from urllib.parse import urlencode
+
+from pymaker import Address, Wad
+from pymaker.util import http_response_summary
+from typing import Optional, List
+
+
+class Order:
+    def __init__(self,
+                 order_id: int,
+                 pair: str,
+                 is_sell: bool,
+                 price: Wad,
+                 amount: Wad):
+
+        assert(isinstance(pair, str))
+        assert(isinstance(is_sell, bool))
+        assert(isinstance(price, Wad))
+        assert(isinstance(amount, Wad))
+
+        self.order_id = order_id
+        self.pair = pair
+        self.is_sell = is_sell
+        self.price = price
+        self.amount = amount
+
+    @property
+    def sell_to_buy_price(self) -> Wad:
+        return self.price
+
+    @property
+    def buy_to_sell_price(self) -> Wad:
+        return self.price
+
+    @property
+    def remaining_buy_amount(self) -> Wad:
+        return self.amount*self.price if self.is_sell else self.amount
+
+    @property
+    def remaining_sell_amount(self) -> Wad:
+        return self.amount if self.is_sell else self.amount*self.price
+
+    def __repr__(self):
+        return pformat(vars(self))
+
+    @staticmethod
+    def create(item):
+        return Order(order_id=item['orderId'],
+                     pair=item['symbol'],
+                     is_sell=True if item['side'] == 'SELL' else False,
+                     price=Wad.from_number(item['price']),
+                     amount=Wad.from_number(item['origQty']))
+
+
+class Trade:
+    def __init__(self,
```

Comment:
Instead of redefining a `Trade` and `Order` class, you can also just import the standard `Trade` and `Order` classes from `pyexchange.model`, and utilize the subclass builder pattern to instantiate a new Order and Trade. You can see an example here: https://github.com/makerdao/pyexchange/blob/master/pyexchange/dydx.py#L52

Reply:
Awesome, I'll use it.